internal/utils/common_utils.go

```diff
@@ -251,10 +254,9 @@ func createBudgetRule(b BudgetRule, gbRule *Rule) monitoringv1.Rule {
 }
 
 func (d DataSourceConfig) ParseTenantAnnotation() (tenants []string) {
-	if len(d.DataSource.Spec.ConnectionDetails.SourceTenants) != 0 {
-		for _, tenant := range d.DataSource.Spec.ConnectionDetails.SourceTenants {
-			tenants = append(tenants, tenant)
-		}
-	}
+	// var sourceTenants []string
+	// for _, tenant := range strings.Split(d.DataSource.Spec.ConnectionDetails["sourceTenants"], ",") {
+	// 	sourceTenants = append(sourceTenants, tenant)
+	// }
 	return tenants
 }
```

Comment:
We have to use the `ConstructConnectionDetails()` func here to get the sourceTenants, somehow?